packages/core/core-flows/src/pricing/steps/create-price-preferences.ts

```diff
@@ -0,0 +1,34 @@
+import { IPricingModuleService } from "@medusajs/types"
+import { PricingWorkflow } from "@medusajs/types/dist/workflow"
+import { ModuleRegistrationName } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type StepInput = PricingWorkflow.CreatePricePreferencesWorkflowInput[]
+
+export const createPricePreferencesStepId = "create-price-preferences"
+export const createPricePreferencesStep = createStep(
+  createPricePreferencesStepId,
+  async (data: StepInput, { container }) => {
+    const pricingModule = container.resolve<IPricingModuleService>(
+      ModuleRegistrationName.PRICING
+    )
+
+    const pricePreferences = await pricingModule.createPricePreferences(data)
+
+    return new StepResponse(
+      pricePreferences,
+      pricePreferences.map((pricePreference) => pricePreference.id)
+    )
+  },
+  async (pricePreferences, { container }) => {
+    if (!pricePreferences?.length) {
+      return
+    }
+
+    const pricingModule = container.resolve<IPricingModuleService>(
+      ModuleRegistrationName.PRICING
+    )
+
+    await pricingModule.deletePricePreferences(pricePreferences)
+  }
```

Comment:
```suggestion
  async (ids, { container }) => {
    if (!ids?.length) {
      return
    }

    const pricingModule = container.resolve<IPricingModuleService>(
      ModuleRegistrationName.PRICING
    )

    await pricingModule.deletePricePreferences(ids)
  }
```

Reply:
damn the automerge was faster 😄 I'll open a new PR now anyway and address this.